tlsdialer.go

```diff
@@ -170,13 +172,23 @@ func (d *Dialer) DialForTimings(network, addr string) (*ConnWithTimings, error)
 	}
 	log.Tracef("ServerName is: %s", serverName)
 
+	validateName := serverName
+	if d.ForceValidateName != "" {
+		validateName = d.ForceValidateName
+	}
+
 	log.Trace("Copying config so that we can tweak it")
 	configCopy := new(tls.Config)
 	*configCopy = *config
 
 	if d.SendServerName {
-		log.Tracef("Setting ServerName to %s and relying on the usual logic in tls.Conn.Handshake() to do verification", serverName)
 		configCopy.ServerName = serverName
+		if serverName == validateName {
+			log.Tracef("Setting ServerName to %s and relying on the usual logic in tls.Conn.Handshake() to do verification", serverName)
```

Comment:
Not new to this PR, but I wonder what's the benefit of verifying via stdlib over the DIY approach. The logic is there anyway, and if the code breaks, we want it to break consistently regardless of the server name and validate name.

Reply:
That's fair... less paths if we just use `verifyServerCerts` in all cases and probably makes any issues with it more obvious.

Reply:
@joesis updated to always use verifyServerCerts, take another look when you have a chance.

Reply:
@ox any reason not to here? The only difference according to the docs is that the tls.Conn's VerifiedChains are not populated in this case (you have to use the structure we send back).

It seems more consistent this way anyway since there were some cases where it was not populated in the prior version (and so we likely shouldn't be counting on it)

Reply:
@forkner My one concern would be that the verification logic doesn't work if we're not explicit about the certs, but it turns out that it does work. I added `TestSimulatedMITMDialingPublicSite` to verify this. So, I think we're good.